clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/orchestration/AsgNameToServerGroupNameDescriptionPreProcessor.groovy

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2016 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+package com.netflix.spinnaker.clouddriver.aws.orchestration
+
+import com.netflix.spinnaker.clouddriver.orchestration.AtomicOperationDescriptionPreProcessor
+import org.springframework.core.annotation.Order
+import org.springframework.stereotype.Component
+
+@Order(-1)
+@Component
+class AsgNameToServerGroupNameDescriptionPreProcessor implements AtomicOperationDescriptionPreProcessor {
+  @Override
+  boolean supports(Class descriptionClass, Map description) {
+    return description.containsKey("asgName")
+  }
+
+  @Override
+  Map process(Map description) {
+    description.serverGroupName = description.serverGroupName ?: description.asgName
+    return description
```

Comment:
add a `description.remove('asgName')` (I guess that means tracking down any usage of asgName and replacing it with serverGroupName)